### DISCUSSION TITLE
buefy-next use i18n in modal with local scope

### DISCUSSION THREAD

**Discussion by wesdevpro** [buefy/buefy#4092] [Q&A] [answered]
Hi, I'm using the current version of buefy-next with nuxt3 (3.8), so  far working pretty good. :)
Is there a way to use i18n with local scope inside the modal ?
I can provide the t object as prop, but then I only have the scope of the parent, not the local scope of the modal. 
If I add `useI18n({useScope: 'local'})` in the async setup routine (using options api, not composition api) I get the error that the modal isn't wrapped in a suspense boundary:
```
setup function returned a promise, but no <Suspense> boundary was found in the parent component tree. A component with async setup() must be nested in a <Suspense> in order to be rendered.
```
but the modal is being called programmatically:
```
this.$buefy.modal.open({
        component: MyTestModal,
        props: {
          form: { a: "test" },
        },
        hasModalCard: false,
        width: 640,
        canCancel: ["x"],
      });
```
Inside the modal:
```
export default defineComponent({
  async setup() {
    const { t } = useI18n({ useScope: 'local' })
    return { t };
  },
  props: {
    ...
```
What acutally **is working** is the integration of the parent object, but this doesn't give me a local scope, only the one of the parent and it doesn't solve the problem with async setup, has to be "normal":
```
export default defineComponent({
  setup() {
    const { t } = useNuxtApp().$i18n;
    return { t };
  },
  props: {
    ....
```

How can I embed i18n inside the setup routine of the modal ?

**The problem seems to be that the modal can't handle the async setup function when called programmatically.**  :)
This also makes fetching graphql queries inside the modal (async setup) impossible. When will this be fixed ?

@mpgalaxy

**Comment by kikuomax** [comment]
@mpgalaxy The problem is twofold.

1. Regarding `Suspense`, I do not know whether `Modal` should take care of async setup or async components. If you wrap your `MyTestModal` in a `Suspense`, let's say `MyTestModalSuspense`, `Modal` can programmatically be opened with `MyTestModalSuspense`.

    ```vue
    <template>
      <Suspense>
        <MyTestModal />
        <template #fallback>loading ...</template>
      </Suspense>
    </template>
    ```

2. I found a more serious problem that `useI18n` does not work inside a setup script within the context of a programmatically opened `Modal`. You will face an error similar to the following:

    ```
    Unhandled Promise Rejection: SyntaxError: Need to install with `app.use` function
    ```

**Reply by kikuomax** [reply, marked as answer]
See
- #153
